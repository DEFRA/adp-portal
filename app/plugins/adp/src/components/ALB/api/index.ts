export { ArmsLengthBodyClient } from './AlbClient'
export type {ArmsLengthBodyApi} from './AlbApi'
