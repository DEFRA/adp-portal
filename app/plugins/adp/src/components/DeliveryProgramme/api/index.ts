export { DeliveryProgrammeClient } from './DeliveryProgrammeClient';
export type { DeliveryProgrammeApi } from './DeliveryProgrammeApi';
