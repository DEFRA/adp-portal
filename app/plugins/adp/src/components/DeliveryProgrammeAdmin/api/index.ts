export { deliveryProgrammeAdminApiRef } from './DeliveryProgrammeAdminApi';
export { DeliveryProgrammeAdminApiClient } from './DeliveryProgrammeAdminClient';
