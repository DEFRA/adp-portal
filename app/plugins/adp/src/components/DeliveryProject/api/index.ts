export * from './DeliveryProjectClient';
export * from './DeliveryProjectApi';
