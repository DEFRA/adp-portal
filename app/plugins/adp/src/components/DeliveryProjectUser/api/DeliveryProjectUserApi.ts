import type {
  CreateDeliveryProjectUserRequest,
  DeliveryProjectUser,
} from '@internal/plugin-adp-common';
import { createApiRef } from '@backstage/core-plugin-api';

export const deliveryProjectUserApiRef = createApiRef<DeliveryProjectUserApi>({
  id: 'plugin.adp.deliveryprojectuserapi',
});

export interface DeliveryProjectUserApi {
  getAll(): Promise<DeliveryProjectUser[]>;
  getByDeliveryProjectId(
    deliveryProjectId: string,
  ): Promise<DeliveryProjectUser[]>;
  create(data: CreateDeliveryProjectUserRequest): Promise<DeliveryProjectUser>;
}
