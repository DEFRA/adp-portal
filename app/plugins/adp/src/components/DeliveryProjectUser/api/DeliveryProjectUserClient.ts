import type {
  DeliveryProjectUser,
  CreateDeliveryProjectUserRequest,
  UpdateDeliveryProjectUserRequest,
  DeleteDeliveryProjectUserRequest,
} from '@internal/plugin-adp-common';
import type { DeliveryProjectUserApi } from './DeliveryProjectUserApi';
import type { DiscoveryApi, FetchApi } from '@backstage/core-plugin-api';
import { ResponseError } from '@backstage/errors';
import { ValidationError } from '../../../utils/ValidationError';

export class DeliveryProjectUserClient implements DeliveryProjectUserApi {
  readonly #discoveryApi: DiscoveryApi;
  readonly #fetchApi: FetchApi;

  constructor(discoveryApi: DiscoveryApi, fetchApi: FetchApi) {
    this.#discoveryApi = discoveryApi;
    this.#fetchApi = fetchApi;
  }

  async getAll(): Promise<DeliveryProjectUser[]> {
    const url = await this.#getBaseUrl();

    const response = await this.#fetchApi.fetch(url);

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    const deliveryProjectUsers = asProjectUsers(await response.json());

    for (const item of deliveryProjectUsers)
      item.updated_at = new Date(item.updated_at);

    return deliveryProjectUsers;
  }

  async getByDeliveryProjectId(
    deliveryProjectId: string,
  ): Promise<DeliveryProjectUser[]> {
    const baseUrl = await this.#getBaseUrl();
    const url = `${baseUrl}/${deliveryProjectId}`;
    const response = await this.#fetchApi.fetch(url);

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    const deliveryProgrammeAdmins = asProjectUsers(await response.json());

    return deliveryProgrammeAdmins;
  }

  async create(
    data: CreateDeliveryProjectUserRequest,
  ): Promise<DeliveryProjectUser> {
    const url = await this.#getBaseUrl();

    const response = await this.#fetchApi.fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    });

    return await this.#handleCreateUpdateResponse(response);
  }

  async update(
    data: UpdateDeliveryProjectUserRequest,
  ): Promise<DeliveryProjectUser> {
    const url = await this.#getBaseUrl();

    const response = await this.#fetchApi.fetch(url, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    });

    return await this.#handleCreateUpdateResponse(response);
  }

  async #getBaseUrl(): Promise<string> {
    return `${await this.#discoveryApi.getBaseUrl('adp')}/deliveryProjectUsers`;
  }

  async #handleCreateUpdateResponse(
    response: Response,
  ): Promise<DeliveryProjectUser> {
    if (response.status === 400)
      throw new ValidationError((await response.json()).errors);

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    return asProjectUser(await response.json());
  }

  async delete(
    deliveryProjectUserId: string,
    deliveryProjectId: string,
  ): Promise<void> {
    const url = await this.#getBaseUrl();

    const body: DeleteDeliveryProjectUserRequest = {
      delivery_project_user_id: deliveryProjectUserId,
      delivery_project_id: deliveryProjectId,
    };

    const response = await this.#fetchApi.fetch(url, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(body),
    });

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }
  }
}

function asProjectUsers(result: DeliveryProjectUser[]) {
  return result.map(asProjectUser);
}

function asProjectUser(result: DeliveryProjectUser) {
  result.updated_at = new Date(result.updated_at);
  return result;
}
