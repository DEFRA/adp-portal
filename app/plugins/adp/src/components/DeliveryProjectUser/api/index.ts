export {
  deliveryProjectUserApiRef,
  type DeliveryProjectUserApi,
} from './DeliveryProjectUserApi';
export { DeliveryProjectUserClient } from './DeliveryProjectUserClient';
