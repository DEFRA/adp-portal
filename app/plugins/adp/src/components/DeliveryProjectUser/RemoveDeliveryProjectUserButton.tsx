import React, { useState } from 'react';
import { Button } from '@material-ui/core';
import type { DeliveryProjectUser } from '@internal/plugin-adp-common';
import { deliveryProjectUserDeletePermission } from '@internal/plugin-adp-common';
import { deliveryProjectUserApiRef } from './api';
import { alertApiRef, useApi } from '@backstage/core-plugin-api';
import { usePermission } from '@backstage/plugin-permission-react';
import { ConfirmationDialog } from '../../utils';

export type RemoveDeliveryProjectUserButtonProps = Readonly<
  Omit<Parameters<typeof Button>[0], 'onClick'> & {
    deliveryProjectUser: DeliveryProjectUser;
    onRemoved?: () => void;
  }
>;

export function RemoveDeliveryProjectUserButton({
  deliveryProjectUser,
  onRemoved,
  ...buttonProps
}: RemoveDeliveryProjectUserButtonProps) {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const alertApi = useApi(alertApiRef);
  const client = useApi(deliveryProjectUserApiRef);
  const { allowed: canRemoveProjectUser } = usePermission({
    permission: deliveryProjectUserDeletePermission,
    resourceRef: deliveryProjectUser.id,
  });

  if (!canRemoveProjectUser) return null;

  async function handleDelete() {
    await client.delete(
      deliveryProjectUser.id,
      deliveryProjectUser.delivery_project_id,
    );
    onRemoved?.();
    alertApi.post({
      message: `Removed ${deliveryProjectUser.name} from this delivery project`,
      display: 'transient',
      severity: 'success',
    });
  }

  return (
    <>
      <Button
        {...buttonProps}
        onClick={() => setIsModalOpen(true)}
        data-testid="remove-delivery-project-user-button"
      >
        Remove
      </Button>

      {isModalOpen && (
        <ConfirmationDialog
          title={`Remove ${deliveryProjectUser.name}?`}
          content="Are you sure you want to remove this user? The user will no longer be able to perform certain actions on the ADP portal. You can re-add the user after removing them."
          confirm="Remove"
          confirmColour="secondary"
          completed={submitted => {
            setIsModalOpen(false);
            if (submitted) onRemoved?.();
          }}
          submit={handleDelete}
        />
      )}
    </>
  );
}
