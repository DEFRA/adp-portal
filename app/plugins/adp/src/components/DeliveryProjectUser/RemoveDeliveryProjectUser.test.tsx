import type { AlertApi } from '@backstage/core-plugin-api';
import { alertApiRef } from '@backstage/core-plugin-api';
import React from 'react';
import type { DeliveryProjectUserApi } from './api';
import { deliveryProjectUserApiRef } from './api';
import type { RemoveDeliveryProjectUserButtonProps } from './RemoveDeliveryProjectUserButton';
import { RemoveDeliveryProjectUserButton } from './RemoveDeliveryProjectUserButton';
import { render, waitFor } from '@testing-library/react';
import { TestApiProvider } from '@backstage/test-utils';
import { SnapshotFriendlyStylesProvider } from '../../utils';
import type { DeliveryProjectUser } from '@internal/plugin-adp-common';
import type * as PluginPermissionReactModule from '@backstage/plugin-permission-react';
import type * as ConfirmationDialogModule from '../../utils/ConfirmationDialog';
import userEvent from '@testing-library/user-event';

function setup() {
  const mockAlertApi: jest.Mocked<AlertApi> = {
    alert$: jest.fn(),
    post: jest.fn(),
  };
  const mockProjectUserApi: jest.Mocked<DeliveryProjectUserApi> = {
    create: jest.fn(),
    delete: jest.fn(),
    getAll: jest.fn(),
    getByDeliveryProjectId: jest.fn(),
    update: jest.fn(),
  };

  return {
    mockAlertApi,
    mockProjectUserApi,
    async renderComponent(props: RemoveDeliveryProjectUserButtonProps) {
      const result = render(
        <TestApiProvider
          apis={[
            [alertApiRef, mockAlertApi],
            [deliveryProjectUserApiRef, mockProjectUserApi],
          ]}
        >
          <SnapshotFriendlyStylesProvider>
            <RemoveDeliveryProjectUserButton {...props} />
          </SnapshotFriendlyStylesProvider>
        </TestApiProvider>,
      );
      await waitFor(() => expect(result.baseElement).not.toBeEmptyDOMElement());
      return { result };
    },
  };
}

const deliveryProjectUser: DeliveryProjectUser = {
  aad_entity_ref_id: 'cffdf0da-48b9-40d0-a519-d5d818b10a84',
  delivery_project_id: 'da61ccb1-95f7-4cff-9b75-28e3aec63e4b',
  email: 'test@test.com',
  id: '896061ec-58dd-468f-bfbc-c35ccb177f36',
  is_admin: true,
  is_technical: false,
  name: 'test user',
  updated_at: new Date(),
  github_username: 'test_user',
};

const usePermission: jest.MockedFn<
  typeof PluginPermissionReactModule.usePermission
> = jest.fn();
const ConfirmationDialog: jest.MockedFn<
  typeof ConfirmationDialogModule.ConfirmationDialog
> = jest.fn();

jest.mock(
  '@backstage/plugin-permission-react',
  () =>
    ({
      get usePermission() {
        return usePermission;
      },
      get IdentityPermissionApi(): never {
        throw new Error('Not mocked');
      },
      get PermissionedRoute(): never {
        throw new Error('Not mocked');
      },
      get RequirePermission(): never {
        throw new Error('Not mocked');
      },
      get permissionApiRef(): never {
        throw new Error('Not mocked');
      },
    }) satisfies typeof PluginPermissionReactModule,
);

jest.mock(
  '../../utils/ConfirmationDialog',
  () =>
    ({
      get ConfirmationDialog() {
        return ConfirmationDialog as typeof ConfirmationDialogModule.ConfirmationDialog;
      },
    }) satisfies typeof ConfirmationDialogModule,
);

describe('RemoveDeliveryProjectUserButton', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should not render when the user is not allowed to remove delivery project users', async () => {
    const { mockAlertApi, mockProjectUserApi, renderComponent } = setup();
    usePermission.mockReturnValue({ allowed: false, loading: false });

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectUser: deliveryProjectUser,
    });

    expect(result.baseElement).toMatchSnapshot();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.delete).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockProjectUserApi.update).not.toHaveBeenCalled();
  });

  it('should only render a button initially', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    usePermission.mockReturnValue({ allowed: true, loading: false });

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectUser: deliveryProjectUser,
    });

    expect(result.baseElement).toMatchSnapshot();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.delete).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockProjectUserApi.update).not.toHaveBeenCalled();
  });

  it('should render the dialog when the button is clicked', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    ConfirmationDialog.mockReturnValue(<span>Test dialog</span>);
    usePermission.mockReturnValue({ allowed: true, loading: false });

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectUser: deliveryProjectUser,
    });
    await userEvent.click(
      result.getByTestId('remove-delivery-project-user-button'),
    );

    expect(result.baseElement).toMatchSnapshot();
    expect(ConfirmationDialog.mock.calls).toHaveLength(1);
    const formProps = ConfirmationDialog.mock.calls[0][0];
    expect({
      open: formProps.open,
      title: formProps.title,
      content: formProps.content,
      confirm: formProps.confirm,
      cancel: formProps.cancel,
    }).toMatchObject({
      open: undefined,
      title: `Remove ${deliveryProjectUser.name}?`,
      content:
        'Are you sure you want to remove this user? The user will no longer be able to perform certain actions on the ADP portal. You can re-add the user after removing them.',
      confirm: 'Remove',
      cancel: undefined,
    });
    expect(formProps.submit).toBeDefined();
    expect(formProps.completed).toBeDefined();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.delete).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockProjectUserApi.update).not.toHaveBeenCalled();
  });

  it('should close the dialog via the Cancel action', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    ConfirmationDialog.mockReturnValue(<span>Test dialog</span>);
    usePermission.mockReturnValue({ allowed: true, loading: false });

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectUser: deliveryProjectUser,
    });
    await userEvent.click(
      result.getByTestId('remove-delivery-project-user-button'),
    );

    expect(result.baseElement).toMatchSnapshot('Before cancel');
    expect(ConfirmationDialog.mock.calls).toHaveLength(1);
    const formProps = ConfirmationDialog.mock.calls[0][0];
    React.act(() => formProps.completed(undefined));
    await waitFor(() => expect(result.queryByText('Test dialog')).toBeNull());
    expect(result.baseElement).toMatchSnapshot('After cancel');
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.delete).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockProjectUserApi.update).not.toHaveBeenCalled();
  });

  it('should call onRemoved when the dialog is closed via the Remove action', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    ConfirmationDialog.mockReturnValue(<span>Test dialog</span>);
    usePermission.mockReturnValue({ allowed: true, loading: false });
    const onRemoved: jest.MockedFn<
      Exclude<RemoveDeliveryProjectUserButtonProps['onRemoved'], undefined>
    > = jest.fn();

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectUser: deliveryProjectUser,
      onRemoved,
    });
    await userEvent.click(
      result.getByTestId('remove-delivery-project-user-button'),
    );

    expect(result.baseElement).toMatchSnapshot('Before complete');
    expect(ConfirmationDialog.mock.calls).toHaveLength(1);
    expect(onRemoved).not.toHaveBeenCalled();
    const formProps = ConfirmationDialog.mock.calls[0][0];
    React.act(() => formProps.completed(true));
    await waitFor(() => expect(result.queryByText('Test dialog')).toBeNull());
    expect(result.baseElement).toMatchSnapshot('After complete');
    expect(onRemoved).toHaveBeenCalledTimes(1);
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.delete).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockProjectUserApi.update).not.toHaveBeenCalled();
  });

  it('should call the API when submitting', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    ConfirmationDialog.mockReturnValue(<span>Test dialog</span>);
    usePermission.mockReturnValue({ allowed: true, loading: false });

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectUser: deliveryProjectUser,
    });
    await userEvent.click(
      result.getByTestId('remove-delivery-project-user-button'),
    );

    expect(result.baseElement).toMatchSnapshot();
    expect(ConfirmationDialog.mock.calls).toHaveLength(1);
    const formProps = ConfirmationDialog.mock.calls[0][0];
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.delete).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockProjectUserApi.update).not.toHaveBeenCalled();

    await formProps.submit();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.delete.mock.calls).toMatchObject([
      [deliveryProjectUser.id, deliveryProjectUser.delivery_project_id],
    ]);
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockProjectUserApi.update).not.toHaveBeenCalled();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post.mock.calls).toMatchObject([
      [
        {
          message: `Removed ${deliveryProjectUser.name} from this delivery project`,
          severity: 'success',
          display: 'transient',
        },
      ],
    ]);
  });
});
