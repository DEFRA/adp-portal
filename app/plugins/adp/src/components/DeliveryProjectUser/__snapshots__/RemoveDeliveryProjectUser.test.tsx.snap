// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveDeliveryProjectUserButton should call onRemoved when the dialog is closed via the Remove action: After complete 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="remove-delivery-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</body>
`;

exports[`RemoveDeliveryProjectUserButton should call onRemoved when the dialog is closed via the Remove action: Before complete 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="remove-delivery-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;

exports[`RemoveDeliveryProjectUserButton should call the API when submitting 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="remove-delivery-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;

exports[`RemoveDeliveryProjectUserButton should close the dialog via the Cancel action: After cancel 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="remove-delivery-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</body>
`;

exports[`RemoveDeliveryProjectUserButton should close the dialog via the Cancel action: Before cancel 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="remove-delivery-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;

exports[`RemoveDeliveryProjectUserButton should not render when the user is not allowed to remove delivery project users 1`] = `
<body>
  <div />
</body>
`;

exports[`RemoveDeliveryProjectUserButton should only render a button initially 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="remove-delivery-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`RemoveDeliveryProjectUserButton should render the dialog when the button is clicked 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="remove-delivery-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Remove
      </span>
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;
